### PR TITLE
Rmb 326 upload complete state set before end of stream

### PR DESCRIPTION
**Pull request by adamdickmeiss**

### CODE REVIEW

File: sample2/src/test/java/org/folio/rest/impl/APITests.java [hunk 1]
```diff
@@ -108,4 +112,34 @@ public void testEchoWithBadLanguage2(TestContext context) {
         .contentType("text/plain")
         .statusCode(400);
   } 
+
+  @Test
+  public void testUpload1(TestContext context) {
```
Comment (julianladisch) [inline sample2/src/test/java/org/folio/rest/impl/APITests.java]: Can this renamed to `testChunkedUpload`?

File: sample2/src/test/java/org/folio/rest/impl/APITests.java [hunk 2]
```diff
@@ -108,4 +112,34 @@ public void testEchoWithBadLanguage2(TestContext context) {
...
+    Async async = context.async();
+    final int sz = 100000;
```
Comment (julianladisch) [inline sample2/src/test/java/org/folio/rest/impl/APITests.java]: Can this renamed to `chunkSize`?

File: sample2/src/test/java/org/folio/rest/impl/APITests.java [hunk 3]
```diff
@@ -108,4 +112,34 @@ public void testEchoWithBadLanguage2(TestContext context) {
...
+    final int cnt = 100;
```
Comment (julianladisch) [inline sample2/src/test/java/org/folio/rest/impl/APITests.java]: Can this renamed to chunkCount?

**Comment by adamdickmeiss** [comment]
I did the renames.. PR failure(s) that seems to be unrelated to THIS work..

**Pull request merged**